### ISSUE TITLE
Add support for Log4j2

### ISSUE THREAD

**Issue by sdeleuze** [spring-projects-experimental/spring-native#115]
Related to #84.

**Comment by sdeleuze** [comment]
See related https://github.com/oracle/graal/issues/808 closed issue.

**Comment by sdeleuze** [comment]
Blocked on oracle/graal#2991.

**Comment by sdeleuze** [comment]
Should be doable with GraalVM 21.1, see [this comment](https://github.com/oracle/graal/issues/2991#issuecomment-787828477) for more details.

**Comment by sdeleuze** [comment]
Some hints are still missing so I reopen this issue.

**Comment by sdeleuze** [comment]
Ok this one is more involved than expected since Log4j2 is really using a log of reflection. I have tried to [craft custom hints](https://github.com/sdeleuze/spring-native/commit/6dc5242434a41b6dad1d01f8848ea0aa3702a1fe) but even with that this is not super maintainable/complete and would require [native support on Log4j2 side](https://issues.apache.org/jira/browse/LOG4J2-2604) (preferred solution) or a dedicated `TypeProcessor` or `NativeConfiguration`. This would be doable but I think we have other priorities tight now, so I move this issue to the backlog, just keeping [`Log4jCoreHints`](https://github.com/spring-projects-experimental/spring-native/blob/main/spring-native-configuration/src/main/java/org/apache/logging/log4j/core/Log4jCoreHints.java) to avoid a hard breakage during compilation with some dependencies like Elasticsearch (see #398).

**Comment by sdeleuze** [comment]
Superseded by https://issues.apache.org/jira/browse/LOG4J2-2649?focusedCommentId=17610488&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-17610488.
